moto/sqs/models.py

```diff
@@ -758,6 +763,24 @@ def send_message(
 
         queue = self.get_queue(queue_name)
 
+        if (
```

Comment:
Just a nitpick - I think these two checks should be behind an `is fifo` check.
The current approach works, because `ContentBasedDeduplication` is only set for `fifo`-queues. In other words: `queue.attributes.get("ContentBasedDeduplication")` always returns None for a non-fifo queue. 

But it's more readable IMO if we make it explicit that these checks are only applicable for FIFO queues.

Reply:
Makes sense

Reply:
Updated the PR